src/plugins/createPluginManager.ts

```diff
@@ -0,0 +1,28 @@
+import {Plugin, PluginSlots} from '@plugins/PluginsContext';
+
+export class PluginManager {
+  private plugins: Plugin[];
+
+  public constructor() {
+    this.plugins = [];
+  }
+
+  public add(plugin: Plugin): void {
+    plugin.order ? this.plugins.splice(plugin.order, 0, plugin) : this.plugins.push(plugin);
+  }
+
+  public setup(): PluginSlots {
+    const pluginSlots: PluginSlots = {
```

Comment:
I wouldn't call these "slots" slots, these are rather some "items".

Besides, there are multiple cases to handle:
* Slots with a single item
* Slots with an array of items
* Data item
* Data lists
* Maybe: data object

Reply:
for now, I just made a distinction between single items and array of items. Only difference for array, is that we need the helper function `insertPluginsToArray`, in order to insert plugin items between default components

Reply:
Plugin mechanism is basically an architecture foundation; it's better to have everything analyzed, covered and tested immediately - otherwise, we would need to anyway immediately do it. If foundation has some flaws and is used already throughout the application, it ends up with complex refactor - many moving parts that have to be redesigned again too